providers/fips/fipsprov.c

```diff
@@ -779,6 +779,18 @@ void ERR_vset_error(int lib, int reason, const char *fmt, va_list args)
     c_vset_error(NULL, ERR_PACK(lib, 0, reason), fmt, args);
 }
 
+void *ERR_raise_malloc_failure(void *ptr,
+                               const char *file, int line, const char *func)
+{
+    if (ptr != NULL)
+        return ptr; /* all good */
+
+    ERR_new();
+    ERR_set_debug(file, line, func);
+    ERR_set_error(ERR_LIB_CRYPTO, ERR_R_MALLOC_FAILURE, NULL);
```

Comment:
This copy of the function should have an argument `int lib` as well, to be used instead of `ERR_LIB_CRYPTO`.

Reply:
As answered above, this would require providing the library information via the `OPENSSL_malloc()` macro and friends.
Certainly this would be too much of a hassle to do explicitly for each and every mem alloc call.
We might also add some OPENSSL_LIB macro that gets defined per crypto module with `ERR_LIB_CRYPTO` being the fallback, but I don't think it would be worth the effort since the file name, line, and function name is already present.